.pipelines/templates/.builder-release-template-windows.yaml

```diff
@@ -15,15 +15,16 @@ steps:
 #               we will hard-code the task variables SIG_GALLERY_NAME, SIG_IMAGE_NAME, and SIG_IMAGE_VERSION.
 #               Task variable GEN2_SIG_FOR_PRODUCTION is set to True and passed to the following steps.
 #               Built sig will be deleted because it has been converted to VHD, and thus not needed.
+#               Use the suffix $(date +"%y%m%d%H%M%S") to avoid duplication while running Gen 2 production pipelines simultaenously.
   - bash: |
       if [[ $(HYPERV_GENERATION) == "V2" ]]; then \
         m="sigMode"; \
         if [[ -n ${SIG_GALLERY_NAME} && -n ${SIG_IMAGE_NAME_PREFIX} && -n ${SIG_IMAGE_VERSION} ]]; then \   
           sigImageName="${SIG_IMAGE_NAME_PREFIX}-${WINDOWS_SKU}-gen2"; \
           echo "##vso[task.setvariable variable=GEN2_SIG_FOR_PRODUCTION]False"; \
         else
-          sigImageName="windows-${WINDOWS_SKU}"; \
-          sigGalleryName="WS2022Gen2Gallery$(date +"%y%m%d")"; \          
+          sigImageName="windows-${WINDOWS_SKU}-gen2-$RANDOM"; \
+          sigGalleryName="WS2022Gen2Gallery$(date +"%y%m%d%H%M%S")$RANDOM"; \          
```

Comment:
Why do you add $RANDOM back? Will it not fail when using $RANDOM?

Reply:
Based on the packer github issue, the same managed image name `windows-$WINDOWS_SKU-gen2` will also fail. I was just trying to see if under absolutely different `sigImageName` and `sigGalleryName`, will concurrent builds fail again.

Even if the gallery name contains $RANDOM as before, I suspect the same `sigImageName`: `windows-2022-containerd-gen2` might also trigger the error in Packer.

Reply:
Replied above.